daemon/logger/syslog/syslog.go

```diff
@@ -55,3 +69,25 @@ func (s *Syslog) Name() string {
 func (s *Syslog) GetReader() (io.Reader, error) {
 	return nil, logger.ReadLogsNotSupported
 }
+
+func parseAddress(address string) (string, string, error) {
+	if urlutil.IsTransportURL(address) {
+		url, err := url.Parse(address)
+		if err != nil {
+			return "", "", err
+		}
+
+		proto := url.Scheme
+		address := url.Host
+		if _, _, err := net.SplitHostPort(address); err != nil {
+			if !strings.Contains(err.Error(), "missing port in address") {
```

Comment:
yep could be another way of checking this but it won't be as readable/simple as this simple string check (and I hate string checking errors..) wdyt?